### DISCUSSION TITLE
Support Extended Query

### DISCUSSION THREAD

**Discussion by trietphm** [jeroenrinzema/psql-wire#13] [General]
Do you have any plan to support the Extended Query?
https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-EXT-QUERY

I can try implementing it but I don't know the detailed specs to start with.

**Comment by jeroenrinzema** [comment]
Hi :wave:, I recently started to work on the extended query protocol.

The `jackc/pgx` driver uses the extended query protocol and is included within the [tests](https://github.com/jeroenrinzema/psql-wire/blob/main/wire_test.go#L174-L207). Feel free to uncomment the test case and check it out.

I have just pushed my changes to a feature branch ([feat/extended-queries](https://github.com/jeroenrinzema/psql-wire/tree/feat/extended-queries)). Feel free to check it out and open a [discussion](https://github.com/jeroenrinzema/psql-wire/discussions) / [issue](https://github.com/jeroenrinzema/psql-wire/issues) if you would like to work on something.

I am planning to make this feature compatible with the simple query handler allowing to handle complex queries with just a single handler. Extended query handlers and components could be [overridden through options](https://github.com/jeroenrinzema/psql-wire/blob/feat/extended-queries/options.go#L46-L61). I have made an draft for this feature within the feature branch.

**Comment by jeroenrinzema** [comment]
@trietphm I have just opened a draft PR to include support for extended queries. Feel free to take a look.

**Reply by trietphm** [reply]
Great! I definitely will check it soon and give feedback if needed.
